test/generators/typescript/preset/MarshallingPreset.spec.ts

```diff
@@ -0,0 +1,207 @@
+    
+      constructor(input: {
+        stringProp?: string,
+        numberProp?: number,
+        objectProp?: NestedTest,
+      }) {
+        this._stringProp = input.stringProp;
+        this._numberProp = input.numberProp;
+        this._objectProp = input.objectProp;
+      }
+    
+      get stringProp(): string | undefined { return this._stringProp; }
+      set stringProp(stringProp: string | undefined) { this._stringProp = stringProp; }
+    
+      get numberProp(): number | undefined { return this._numberProp; }
+      set numberProp(numberProp: number | undefined) { this._numberProp = numberProp; }
+    
+      get objectProp(): NestedTest | undefined { return this._objectProp; }
+      set objectProp(objectProp: NestedTest | undefined) { this._objectProp = objectProp; }
+    
+      get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null | number> | undefined { return this._additionalProperties; }
+      set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null | number> | undefined) { this._additionalProperties = additionalProperties; }
+    
+      get sTestPatternProperties(): Map<String, string> | undefined { return this._sTestPatternProperties; }
+      set sTestPatternProperties(sTestPatternProperties: Map<String, string> | undefined) { this._sTestPatternProperties = sTestPatternProperties; }
+    
+      public marshal() : string {
+        let json = '{'
+        if(this.stringProp !== undefined) {
+          json += `"stringProp": ${JSON.stringify(this.stringProp)},`; 
+        }
+        if(this.numberProp !== undefined) {
+          json += `"numberProp": ${JSON.stringify(this.numberProp)},`; 
+        }
+        if(this.objectProp !== undefined) {
+          json += `"objectProp": ${this.objectProp.marshal()},`; 
+        }
+    
+        if(this.sTestPatternProperties !== undefined) { 
+          for (const [key, value] of this.sTestPatternProperties.entries()) {
+            //Only render pattern properties which are not already a property
+            if(Object.keys(this).includes(String(key))) continue;
+            json += `"${key}": ${JSON.stringify(value)},`;    
+          }
+        }
+    
+        if(this.additionalProperties !== undefined) { 
+          for (const [key, value] of this.additionalProperties.entries()) {
+            //Only render additionalProperties which are not already a property
+            if(Object.keys(this).includes(String(key))) continue;
+            json += `"${key}": ${JSON.stringify(value)},`;    
+          }
+        }
+        //Remove potential last comma 
+        return `${json.charAt(json.length-1) === ',' ? json.slice(0, json.length-1) : json}}`;
+      }
+    
+      public static unmarshal(json: string | object): Test {
+        const obj = typeof json === "object" ? json : JSON.parse(json);
+        const instance = new Test({});
```

Comment:
What about the case when some properties is required? User will have TS error that we we try to pass the empty object without required properties. To fix this, please change it to:

```suggestion
        const instance = new Test({} as unknown as any);
```

Please also add at least the one required property in the json schema :)

Reply:
Damn, good catch once again 👍 `as any` is enough though.